internal/modules/list.go

```diff
@@ -89,7 +99,168 @@ func List(ctx context.Context, client kyma.Interface) (ModulesList, error) {
 	return modulesList, nil
 }
 
-func getInstallDetails(kyma *kyma.Kyma, releaseMetas kyma.ModuleReleaseMetaList, moduleName string) ModuleInstallDetails {
+func getModuleState(ctx context.Context, client kube.Client, moduleTemplate kyma.ModuleTemplate, kymaCR *kyma.Kyma) (string, error) {
+	// get state from Kyma CR if it exists
+	if state := getStateFromKymaCR(moduleTemplate, kymaCR); state != "" {
+		return state, nil
+	}
+
+	// get state from moduleTemplate.Spec.Data if it exists
+	state, err := getStateFromData(ctx, client, moduleTemplate.Spec.Data)
+	if err != nil {
+		return "", err
+	}
+	if state != "" {
+		return state, nil
+	}
+
+	// get state from resource described in moduleTemplate.Spec.Manager if it exists
+	state, err = getResourceState(ctx, client, moduleTemplate.Spec.Manager)
+	if err != nil {
+		return "", err
+	}
+	if state != "" {
+		return state, nil
+	}
+
+	return "", nil
+}
+
+func getStateFromKymaCR(moduleTemplate kyma.ModuleTemplate, kymaCR *kyma.Kyma) string {
+	if kymaCR != nil {
+		for _, module := range kymaCR.Status.Modules {
+			if module.Name == moduleTemplate.Spec.ModuleName {
+				if module.State != "" {
+					return module.State
+				}
+			}
+		}
+	}
+	return ""
```

Comment:
```suggestion
	if kymaCR == nil {
		return ""
	}
	for _, module := range kymaCR.Status.Modules {
		if module.Name == moduleTemplate.Spec.ModuleName && module.State != "" {
			return module.State
		}
	}
```

Reply:
I'll have to `return ""` anyway, this change would make code less readable